api/preprints/serializers.py

```diff
@@ -255,6 +256,10 @@ def subjects_self_view(self):
         # Overrides TaxonomizableSerializerMixin
         return 'preprints:preprint-relationships-subjects'
 
+    def get_is_latest_version(self, obj):
+        versions = [preprint.version for preprint in obj.get_preprint_versions()]
+        return obj.version == max(versions)
```

Comment:
We need a helper method in the model or mixin, which is called by this API serializer method

```python
def get_is_latest_version(self, obj):
    return obj.is_latest_version()
```

Reply:
As for the latest version check, it's not the version with the max version number, the preprint must be published as well. See ticket description for more info.

Check with Ostap on this, we may not have this piece in yet: if we can guarantee that `guid` always points to the latest preprint `version`, we can skip calling `obj.get_preprint_versions()` which is , but get the latest version from base_guid -> referrant -> version.